python/lsst/dax/ppdb/sql/bulk_insert.py

```diff
@@ -87,7 +88,7 @@ def insert(self, table: sqlalchemy.schema.Table, data: ApdbTableData, *, chunk_s
     @staticmethod
     def _row_to_dict(column_names: Sequence[str], row: tuple, drop_columns: set[str]) -> dict[str, Any]:
         """Convert TableData row into dict."""
-        data = dict(zip(column_names, row))
+        data = dict(zip(column_names, row, strict=True))
```

Comment:
Are you expecting different lengths? Are you okay with some values being dropped (since shortest wins)?

Reply:
With `strict=True`, it should throw an error if the lengths are different, which is the desired behavior.